File: src/main/java/cloud/common/NodeConfiguration.java
package cloud.common;

import instance.Node;
import instance.common.Block;
import se.sics.kompics.address.Address;

import java.io.Serializable;
import java.util.List;
import java.util.Map;

/**
 * 
 * @author Amir Moulavi
 * @date 2011-03-27
 *
 */

public class NodeConfiguration implements Serializable {
	
	private static final long serialVersionUID = -9025878066809528650L;
	private CpuConfiguration cpuConfiguration;
	private BandwidthConfiguration bandwidthConfiguration;
	private int simultaneousDownloads;
	private String name = "NONAME";
	private Node node;
	private List<Block> blocks;
	private int memory;
	private Map<String, Address> dataBlocksMap;

	public NodeConfiguration(double cpuSpeed, double bandwidth, int memory, int simultaneousDownloads) {
		this.cpuConfiguration = new CpuConfiguration(cpuSpeed);
		this.bandwidthConfiguration = new BandwidthConfiguration(bandwidth);
		this.memory = memory;
		this.simultaneousDownloads = simultaneousDownloads;
	}

	public NodeConfiguration() {
		this.cpuConfiguration = new CpuConfiguration(2.0);
		this.bandwidthConfiguration = new BandwidthConfiguration(2.0);
		this.memory = 4;
		this.simultaneousDownloads = 20;
	}

	public int getMemory() {
		return memory;
	}

	public int getSimultaneousDownloads() {
		return simultaneousDownloads;
	}
	
	public void setNodeName(String name) {
		this.name = name;
	}

	public String getName() {
		return name;
	}

	public void setNodeInfo(Node node) {
		this.node = node;
		this.name = node.getNodeName();
	}

	public Node getNode() {
		return node;
	}
	
	public String toString() {
		StringBuilder sb = new StringBuilder();
		sb.append("{");
			sb.append(cpuConfiguration).append(",");
			sb.append("memory: ").append(memory).append(",");
			sb.append(bandwidthConfiguration).append(",");
			sb.append("simDownloads: ").append(simultaneousDownloads);
		sb.append("}");
		return sb.toString();
	}

	public void setBlocks(List<Block> blocks) {
		this.blocks = blocks;		
	}

	public List<Block> getBlocks() {
		return blocks;
	}
	
	public NodeConfiguration cpu(double speed) {
		this.cpuConfiguration = new CpuConfiguration(speed);
		return this;
	}
	
	public NodeConfiguration memoryGB(int size) {
		this.memory = size;
		return this;
	}
	
	public NodeConfiguration bandwidthMB(double bandwidth) {
		this.bandwidthConfiguration = new BandwidthConfiguration(bandwidth);
		return this;
	}

	public void setDataBlocksMap(Map<String, Address> dataBlocksMap) {
		this.dataBlocksMap = dataBlocksMap;
	}

	public Map<String, Address> getDataBlocksMap() {
		return dataBlocksMap;
	}

	public CpuConfiguration getCpuConfiguration() {
		return cpuConfiguration;
	}
	
	public BandwidthConfiguration getBandwidthConfiguration() {
		return bandwidthConfiguration;
	}
	
}

File: src/main/scala/cloud/common/NodeConfiguration.scala
/**
 * Copyright 2011 Amir Moulavi (amir.moulavi@gmail.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package cloud.common

import instance.Node
import se.sics.kompics.address.Address
import instance.common.Block
import reflect.BeanProperty
import akka.actor.ActorRef
import scala.collection.mutable

/**
 * @author Amir Moulavi
 * @date 2011-07-09
 *
 */
class NodeConfiguration(cpuSpeed: Double,
                        bandwidth: Double,
                        @BeanProperty var memory: Int,
                        @BeanProperty var simultaneousDownloads: Int)
  extends Serializable {

  def this() = this(2.0, 2.0, 4, 70)

  @BeanProperty
  var cpuConfiguration: CpuConfiguration = new CpuConfiguration(cpuSpeed)
  @BeanProperty
  var bandwidthConfiguration: BandwidthConfiguration = new BandwidthConfiguration(bandwidth)
  @BeanProperty
  var name: String = "NONAME"
  @BeanProperty
  var node: Node = null
  @BeanProperty
  var blocks = List.empty[Block]
  @Deprecated
  @BeanProperty
  var dataBlocksMap: java.util.Map[String, Address] = null
  @BeanProperty
  var blocksMap = mutable.Map.empty[String, ActorRef]
  @BeanProperty
  var headLess: Boolean = false

  def setNodeInfo(node: Node) {
    this.node = node
    this.name = node.getNodeName
  }

  def cpu(speed: Double): NodeConfiguration = {
    this.cpuConfiguration = new CpuConfiguration(speed)
    this
  }

  def memoryGB(size: Int): NodeConfiguration = {
    this.memory = size
    this
  }

  def bandwidthMB(bandwidth: Double): NodeConfiguration = {
    this.bandwidthConfiguration = new BandwidthConfiguration(bandwidth)
    this
  }

  override def toString: String = {
    val sb: StringBuilder = new StringBuilder
    sb.append("{")
    sb.append("cpuConfig").append(cpuConfiguration).append(",")
    sb.append("memory: ").append(memory).append(",")
    sb.append("bandwidth").append(bandwidthConfiguration).append(",")
    sb.append("simDownloads: ").append(simultaneousDownloads)
    sb.append("}")
    sb.toString
  }
}